eth-node/bridge/geth/wakuv2.go

```diff
@@ -338,14 +306,75 @@ func (w *gethWakuV2Wrapper) ConfirmMessageDelivered(hashes []common.Hash) {
 	w.waku.ConfirmMessageDelivered(hashes)
 }
 
-func (w *gethWakuV2Wrapper) SetStorePeerID(peerID peer.ID) {
-	w.waku.SetStorePeerID(peerID)
-}
-
 func (w *gethWakuV2Wrapper) PeerID() peer.ID {
 	return w.waku.PeerID()
 }
 
-func (w *gethWakuV2Wrapper) PingPeer(ctx context.Context, peerID peer.ID) (time.Duration, error) {
-	return w.waku.PingPeer(ctx, peerID)
+func (w *gethWakuV2Wrapper) GetActiveStorenode() peer.ID {
+	return w.waku.StorenodeCycle.GetActiveStorenode()
+}
+
+func (w *gethWakuV2Wrapper) OnStorenodeAvailableOneShot() <-chan struct{} {
+	return w.waku.StorenodeCycle.StorenodeAvailableOneshotEmitter.Subscribe()
+}
+
+func (w *gethWakuV2Wrapper) OnStorenodeChanged() <-chan peer.ID {
+	return w.waku.StorenodeCycle.StorenodeChangedEmitter.Subscribe()
+}
+
+func (w *gethWakuV2Wrapper) OnStorenodeNotWorking() <-chan struct{} {
+	return w.waku.StorenodeCycle.StorenodeNotWorkingEmitter.Subscribe()
+}
+
+func (w *gethWakuV2Wrapper) OnStorenodeAvailable() <-chan peer.ID {
+	return w.waku.StorenodeCycle.StorenodeAvailableEmitter.Subscribe()
+}
+
+func (w *gethWakuV2Wrapper) WaitForAvailableStoreNode(timeout time.Duration) bool {
+	return w.waku.StorenodeCycle.WaitForAvailableStoreNode(context.TODO(), timeout)
+}
+
+func (w *gethWakuV2Wrapper) SetStorenodeConfigProvider(c history.StorenodeConfigProvider) {
+	w.waku.StorenodeCycle.SetStorenodeConfigProvider(c)
+}
+
+func (w *gethWakuV2Wrapper) ProcessMailserverBatch(
+	ctx context.Context,
+	batch types.MailserverBatch,
+	storenodeID peer.ID,
+	pageLimit uint64,
+	shouldProcessNextPage func(int) (bool, uint64),
+	processEnvelopes bool,
+) error {
+	pubsubTopic := w.waku.GetPubsubTopic(batch.PubsubTopic)
+	contentTopics := []string{}
+	for _, topic := range batch.Topics {
+		contentTopics = append(contentTopics, wakucommon.BytesToTopic(topic.Bytes()).ContentTopic())
+	}
+
+	criteria := store.FilterCriteria{
+		TimeStart:     proto.Int64(int64(batch.From) * int64(time.Second)),
+		TimeEnd:       proto.Int64(int64(batch.To) * int64(time.Second)),
```

Comment:
Would be nice to refactor `MailserverBatch` to have `From`/`To` fields as `time.Time` to avoid such weird conversions. Considering that we do it initially as well:
https://github.com/status-im/status-go/blob/a06eb72c012834100feafc1a90b7d950b1d4d57d/protocol/messenger_mailserver.go#L175-L177

And then we have to convert them once again here, when passing to go-waku...

Reply:
I did this change, but while working on it, i realized the madness of how we handle time in status-go: we use uint64 for clock values (milliseconds), on waku we use time.Time, on mailserver code and message gaps calculation we use uint32 (in seconds, i imagine this is 'inherited' from back when we used to have this logic on the mobile app).

This lack of consistency is awful. In a subsequent PR i'll attempt to get rid of the uint32 code for mailserver code and gaps to use time.Time, but perhaps we should consider to later also attempt to get rid of the milliseconds time (uint64) and use time.Time consistently across all codebase?

Reply:
> use time.Time consistently across all codebase

💯